girder/api/v1/folder.py

```diff
@@ -82,10 +86,41 @@ def find(self, parentType, parentId, text, name, limit, offset, sort):
                 offset=offset, limit=limit, sort=sort, filters=filters)
         elif text:
             return self._model.textSearch(
-                text, user=user, limit=limit, offset=offset, sort=sort)
+                text, user=user, limit=limit, offset=offset, sort=sort, filters=filters)
         else:
             raise RestException('Invalid search mode.')
 
+    @access.public(scope=TokenScope.DATA_READ)
+    @autoDescribeRoute(
+        Description('Report the offset of a folder in a list or search.')
+        .notes('You must pass either a "folderId" or "text" field '
+               'to specify how you are searching for folders.  '
+               'If you omit one of these parameters the request will fail and respond : '
+               '"Invalid search mode."')
+        .modelParam('id', model=FolderModel, level=AccessType.READ)
+        .param('parentType', "Type of the folder's parent", required=False,
+               enum=['folder', 'user', 'collection'])
+        .param('parentId', "The ID of the folder's parent.", required=False)
+        .param('text', 'Pass to perform a text search.', required=False)
+        .param('name', 'Pass to lookup a folder by exact name match. Must '
+               'pass parentType and parentId as well when using this.', required=False)
+        .pagingParams(defaultSort='lowerName')
+        .errorResponse()
+        .errorResponse('Read access was denied on the parent resource.', 403)
+    )
+    def findPosition(self, folder, parentType, parentId, text, name, limit, offset, sort):
+        filters = {}
+        if len(sort) != 1 or sort[0][0] not in folder:
+            raise RestException('Invalid sort mode.')
+        dir = '$lt' if sort[0][1] == SortDir.ASCENDING else '$gt'
+        filters = {'$or': [
+            {sort[0][0]: {dir: folder.get(sort[0][0])}},
+            {sort[0][0]: folder.get(sort[0][0]), '_id': {dir: folder['_id']}}
+        ]}
+        # limit and offset are actually ignored
```

Comment:
Given that these are ignored, I think it might be misleading to include them as documented parameters to the REST API. But then we wouldn't be able to use `pagingParam` as-is. I don't feel very strongly one way or the other.

Reply:
I thought it better to mirror the find endpoint, but I also don't feel strongly about it.

Reply:
Would it be a breaking change to remove these parameters in the future? If we think so, let's go ahead and remove them now. We could do that by adding a `sortParams` on `Description` that is just that subset of `pagingParams`.

Reply:
I've changed it to remove the paging parameters and explicitly add sort and sortdir parameters.